light-client/src/components/scheduler.rs

```diff
@@ -0,0 +1,80 @@
+use crate::prelude::*;
+use contracts::*;
+
+#[contract_trait]
+pub trait Scheduler {
+    #[pre(light_store.latest(VerifiedStatus::Verified).is_some())]
+    #[post(valid_schedule(ret, target_height, next_height, light_store))]
+    fn schedule(
+        &self,
+        light_store: &dyn LightStore,
+        next_height: Height,
+        target_height: Height,
+    ) -> Height;
+}
+
+#[contract_trait]
+impl<F> Scheduler for F
+where
+    F: Fn(&dyn LightStore, Height, Height) -> Height,
+{
+    fn schedule(
+        &self,
+        light_store: &dyn LightStore,
+        next_height: Height,
+        target_height: Height,
+    ) -> Height {
+        self(light_store, next_height, target_height)
+    }
+}
+
+#[pre(light_store.latest(VerifiedStatus::Verified).is_some())]
+#[post(valid_schedule(ret, target_height, next_height, light_store))]
+pub fn schedule(
+    light_store: &dyn LightStore,
```

Comment:
Why do we need to pass the whole store if all we need is the latest trusted height?

Reply:
This is to match the spec, as @josef-widder suggested the scheduler could perform some optimizations as to what height to verify next based on what headers of nearby heights are already available in the store, but this is not implemented yet.